### DISCUSSION TITLE
Check if user has role in other table

### DISCUSSION THREAD

**Discussion by starredev** [pocketbase/pocketbase#3048] [Q&A] [answered]
Hello,

I have a collection modules with a multiple relation field to ldap_roles. In my user collection I have also a multiple field relation to ldap_roles. Now I want to compare modules roles with user roles, if the user exists the specific role in modules.

And I'm checking it in my middleware with the following code

````
/* Check if user is authenticated to the specific route */
const method = req.path.split("/").pop();
const { error, result } = await asyncHandler(req.pbClient.collection('modules').getFirstListItem(`method="${method}"`, { method: method }));
    
/* Return error if permission is not and user will be logged out */
if (error) return res.status(403).send({ message: 'Access denied. Permission not set' });
````

I'm struggling with creating api rule to get module by method and compare the roles with user roles.

I came up with something like: 
````@request.query.method = @collection.modules.method && (@collection.modules.roles.id ?= @collection.users.roles.id)```` 

but unfortunately that doesnt work

Could you help me?

Bryan

**Comment by ganigeorgiev** [comment, marked as answer]
The `@collection.*` should be used only when you want to "join" another collection to the current one (_including itself_).

For this particular use case I don't think you need it. You can define the following `modules` collection API rule:

```js
method = @request.query.method && roles.id ?= @request.auth.roles.id
```

If the above is not what you are looking for, please provide an example data sample and expected output.

**Reply by starredev** [reply]
That works as expected! thank you so much!! 

I Didn't know that I could make the call roles.id, i though i had to do that with @collection
